libbeat/monitoring/report/elasticsearch/client.go

```diff
@@ -79,23 +79,58 @@ func (c *publishClient) Close() error {
 
 func (c *publishClient) Publish(batch publisher.Batch) error {
 	events := batch.Events()
-	bulk := make([]interface{}, 0, 2*len(events))
+	var failed []publisher.Event
+	var reason error
+	var params map[string]string
```

Comment:
we still need the global 'params' variable? It's re-initialized in the loop for every bulk request.

Reply:
no, this resolved also my comment above.